### DISCUSSION TITLE
Docker-compose version?

### DISCUSSION THREAD

**Discussion by jojochuang** [apache/ozone#6239] [FAQ]
Hi I'm curious about the docker-compose version used in the Ozone's GitHub Action. Where does it get defined and what is the current version?

I'm using docker-compose 2.23.3 on Mac, and running ozonesecure-ha (acceptance HA-secure), it gives me this error:

> unknown service "datanode"

This is because docker-compose.yaml under ozonesecure-ha defines datanode1, datanode2 and datanode3 services but there's no datanode service.

To workaround this issue, I had to manually change testlib.sh

`docker-compose --ansi never up -d --scale datanode="${datanode_count}"`

to 

`docker-compose --ansi never up -d`

I suspect this is only happening for newer docker-compose, and that old versions simply ignores undefined services.

**Comment by adoroszlai** [comment]
CI uses Ubuntu 20.04.  Image contains:

Docker Compose v1 1.29.2
Docker Compose v2 2.23.3

https://github.com/actions/runner-images/blob/ubuntu20/20240212.2/images/ubuntu/Ubuntu2004-Readme.md

**Comment by errose28** [comment]
I filed HDDS-9031 a while ago to migrate the acceptance tests to v2. We will need to do some minor modifications due to the unknown service restriction in the newer version.
